lib/gems/pending/appliance_console/key_configuration.rb

```diff
@@ -45,11 +46,15 @@ def ask_question_loop
     end
 
     def activate
+      if key_exist? && force
+        backup_key
+      end
+
```

Comment:
If fetching fail, the fetch_key would return false, success_get_new will be false, so `remove_backup_key_if_any` and `return true` won't execute. Then we will get to the end of if and `restore_key_if_any` so the good backup will replace bad v2_key.
```ruby
      if remove_key(force)
        success_get_new = fetch_key? ? fetch_key : create_key
        if success_get_new
          remove_backup_key_if_any
          return true
        end
      else
        # probably only got here via the cli
        $stderr.puts
        $stderr.puts "Only generate one encryption key (v2_key) per installation."
        $stderr.puts "Chances are you did not want to overwrite this file."
        $stderr.puts "If you do this all encrypted secrets in the database will not be readable."
        $stderr.puts "Please backup your key and run this command again with --force-key."
        $stderr.puts
      end
      restore_key_if_any
```